src/test/java/com/commercetools/sync/sdk2/types/utils/LocalizedEnumValueUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,233 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.commercetools.api.models.type.TypeAddLocalizedEnumValueActionBuilder;
+import com.commercetools.api.models.type.TypeChangeLocalizedEnumValueLabelActionBuilder;
+import com.commercetools.api.models.type.TypeChangeLocalizedEnumValueOrderActionBuilder;
+import com.commercetools.api.models.type.TypeUpdateAction;
+import com.commercetools.sync.sdk2.commons.exceptions.DuplicateKeyException;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class LocalizedEnumValueUpdateActionUtilsTest {
+  private static final String FIELD_NAME_1 = "field1";
+
+  @Test
+  void
+      buildLocalizedEnumUpdateActions_WithEmptyLocalizedEnumValuesAndNoOldEnumValues_ShouldNotBuildActions() {
+    final List<TypeUpdateAction> updateActions =
+        buildLocalizedEnumValuesUpdateActions(FIELD_NAME_1, emptyList(), emptyList());
+
+    assertThat(updateActions).isEmpty();
+  }
+
+  @Test
+  void
+      buildLocalizedEnumUpdateActions_WithNewLocalizedEnumValuesAndNoOldLocalizedEnumValues_ShouldBuild3AddActions() {
+    final List<TypeUpdateAction> updateActions =
+        buildLocalizedEnumValuesUpdateActions(FIELD_NAME_1, emptyList(), ENUM_VALUES_ABC);
+
+    assertThat(updateActions)
+        .containsExactly(
+            TypeAddLocalizedEnumValueActionBuilder.of()
+                .fieldName(FIELD_NAME_1)
+                .value(ENUM_VALUE_A)
+                .build(),
+            TypeAddLocalizedEnumValueActionBuilder.of()
+                .fieldName(FIELD_NAME_1)
+                .value(ENUM_VALUE_B)
+                .build(),
+            TypeAddLocalizedEnumValueActionBuilder.of()
+                .fieldName(FIELD_NAME_1)
+                .value(ENUM_VALUE_C)
+                .build());
+  }
+
+  @Test
+  void buildLocalizedEnumUpdateActions_WithIdenticalLocalizedEnum_ShouldNotBuildUpdateActions() {
+    final List<TypeUpdateAction> updateActions =
+        buildLocalizedEnumValuesUpdateActions(FIELD_NAME_1, ENUM_VALUES_ABC, ENUM_VALUES_ABC);
+
+    assertThat(updateActions).isEmpty();
+  }
+
+  @Test
+  void buildLocalizedEnumUpdateActions_WithOneLocalizedEnumValue_ShouldBuildAddEnumValueAction() {
+    final List<TypeUpdateAction> updateActions =
+        buildLocalizedEnumValuesUpdateActions(FIELD_NAME_1, ENUM_VALUES_ABC, ENUM_VALUES_ABCD);
+
+    assertThat(updateActions)
+        .containsExactly(
+            TypeAddLocalizedEnumValueActionBuilder.of()
+                .fieldName(FIELD_NAME_1)
+                .value(ENUM_VALUE_D)
+                .build());
+  }
+
+  @Test
+  void buildLocalizedEnumUpdateActions_WithOneEnumValueSwitch_ShouldBuildAddEnumValueActions() {
+    final List<TypeUpdateAction> updateActions =
+        buildLocalizedEnumValuesUpdateActions(FIELD_NAME_1, ENUM_VALUES_ABC, ENUM_VALUES_ABD);
+
+    // remove enum value actions not exists for type resources
```

Comment:
Interesting that there is indeed no remove enum action for Types. But [such action](https://docs.commercetools.com/api/projects/productTypes#remove-enumvalues-from-attributedefinition) exists for product type. Do you know why? If not, can you create a support ticket to ask for this?

Reply:
https://commercetools.atlassian.net/servicedesk/customer/portal/22/SUPPORT-21757?created=true